### ISSUE TITLE
Bad Lexical Cast Error when port number is >=32768

### ISSUE THREAD

**Issue by zfkdnop** [ethereum-mining/ethminer#1906]
**Describe the bug**
Using the following parameters:
`ethminer.exe -U -P stratum://scrubbed%40scrubbed%2escrubbed.miner1:x@etc.pool.minergate.com:45777`
I was unable to get ethminer working with my MinerGate account because of the following error:
> ethminer 0.18.0-rc.0
> Build: windows/release/msvc
> Error: Bad URI : bad lexical cast: source type value could not be interpreted as target

Using the following parameters:
`ethminer.exe -U -P stratum://scrubbed%40scrubbed%2escrubbed.miner1:x@etc.pool.minergate.com:4577`
(notice the missing digit at the end '7')
I was able to get ethminer running correctly BUT this is the incorrect port number for MinerGate. The correct port number is 45777

After some troubleshooting I discovered that **ethminer will fail with the above error message when the port number is >=32768**

**To Reproduce**
Using the following paramaters:
`ethminer.exe -U -P stratum://scrubbed%40scrubbed%2escrubbed.miner1:x@etc.pool.minergate.com:**PORT**`
- Change **PORT** to any number <=32767 - the command will succeed and ethminer will attempt to connect
- Change **PORT** to any number >32767 - the command will fail with the following error:
> Error: Bad URI : bad lexical cast: source type value could not be interpreted as target

**Expected behavior**
Acceptable port numbers should be between the standard 0 and 65535

**Environment (please complete the following information):**
- ethminer 0.18.0-rc.0
Build: windows/release/msvc
- Operating System: Windows 10 Home
- System type: 64bit

**Comment by lesjokolat** [comment]
try changing the :x in password to %3ax see if that helps.

**Comment by zfkdnop** [comment]
The password is "x" not ":x", its my understanding that the URL-encoded chars are necessary when you have, for example, .'s and @'s and :'s as part of the input. Using URL-encoded characters is to allow you to use those special characters as part of your input without throwing off the interpretation of the connection string

By using %3a that would change the workername from "miner1" to "miner1:x" and I would then have no password specified

I tried it though, and still receive the same error:
> C:\Users\scrubbed\Desktop\ethminer-0.18.0-rc.0-cuda8.0-windows-amd64\bin>ethminer.exe -U -P stratum://scrubbed%40scrubbed%2escrubbed.miner1%3ax@etc.pool.minergate.com:45777
> 
> ethminer 0.18.0-rc.0
> Build: windows/release/msvc
> Error: Bad URI : bad lexical cast: source type value could not be interpreted as target

Also, btw, if I make the portnumber less than 32768 I do not get the error...

I genuinely believe it has something to do with the port number being like a ..... what is it called? its like the portnumber variable should be a 4byte integer but instead its a 2byte integer or w/e... an unsigned int versus a signed int.. I hope that makes sense. It's as if its only able to store a value up to 32767 and any higher than that it overflows and causes and error.



For clarification, here are the settings I wish to use in my connection string:
user = scrubbed@scrubbed.scrubbed
workername = miner1
password = x
hostname:port = etc.pool.minergate.com:45777

Given the above, I have set my connection string to:
`stratum://scrubbed%40scrubbed%2escrubbed.miner1:x@etc.pool.minergate.com:45777`
